### DISCUSSION TITLE
How to Deploy hugomods template on Netlify？

### DISCUSSION THREAD

**Discussion by AgentFoundation** [hbstack/.github#48] [General]
I want to build a website the same as hugomods,How to Deploy hugomods template on Netlify？

**Comment by AgentFoundation** [comment]
(https://hbstack.dev/docs/deployment/netlify/)I just find this guide，but it dosen't work

**Reply by razonyang** [reply]
Could you please post deploy error logs?

> Navigate to Netlify Dashboard -> Select a site -> _Deploys_ -> Select a deployment.

Btw, the `HUGO_VERSION` on the config maybe out-date, please consider replacing it with latest, e.g. `0.121.2`

**Comment by AgentFoundation** [comment]
11:10:24 AM: WARN  found no layout file for "html" for kind "home": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
11:10:24 AM: WARN  found no layout file for "html" for kind "taxonomy": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.

**Reply by razonyang** [reply]
Seems that you haven't use any theme, does your site work locally (hugo server)?
Could you provide a Git repo for reproducing the issue?

**Reply by AgentFoundation** [reply]
This Git repo：https://github.com/coding-pub/openmodels，thanks

**Reply by razonyang** [reply]
I see, you're cloning the `hugomods/site`, which is a monorepo repo that contains multiple sites. I'd suggest installing it from scratch. 

- https://theme-cards.hbstack.dev/docs/install-from-scratch/: cards style theme.
- https://hbstack.dev/docs/getting-started/installation/: starter theme, you can switch to cards style theme from it.

If you want to host the hugomods/site for some reason, you should change the build `command` and `publish` on `netlify.toml`, because the source code of main site was put under the `main` folder.

```toml
[build]
command = "cd main && npm ci && hugo --minify --gc -b $URL"
publish = "main/public"
```

You'll need to remove the `baseURL` from `languages.yaml`, I believe that you will need to do extra work to set up multihost on Netlify, please Googling it about this.

https://github.com/coding-pub/openmodels/blob/52363a6fe5c8c2015b993333ba5a51952bb9b9cd/main/config/_default/languages.yaml#L4
https://github.com/coding-pub/openmodels/blob/52363a6fe5c8c2015b993333ba5a51952bb9b9cd/main/config/_default/languages.yaml#L9
